### PR TITLE
Change matrices and other fixes for python bindings

### PR DESCRIPTION
**Pull request by teseoch**
[Describe your changes and what you've already done to test it.]

#### Check all that apply (change to `[x]`)
- [x] All changes meet [libigl style-guidelines](https://libigl.github.io/style-guidelines/).
- [ ] Adds new .cpp file.
- [ ] Adds corresponding unit test.
- [ ] Adds corresponding python binding.
- [x] This is a minor change.

### CODE REVIEW

File: include/igl/bfs.cpp [hunk 1]
```diff
@@ -59,7 +59,7 @@ template <
   typename DType,
   typename PType>
 IGL_INLINE void igl::bfs(
-  const Eigen::SparseMatrix<AType> & A,
+  const AType & A,
```
Comment (fwilliams) [inline include/igl/bfs.cpp]: Are we sure we can't use `SparseCompressedBase`?

File: include/igl/bfs.h [hunk 2]
```diff
@@ -42,7 +42,7 @@ namespace igl
     typename DType,
     typename PType>
   IGL_INLINE void bfs(
-    const Eigen::SparseMatrix<AType> & A,
+    const AType & A,
```
Comment (fwilliams) [inline include/igl/bfs.h]: Same comment as above

File: include/igl/arap_linear_block.h [hunk 3]
```diff
@@ -43,32 +43,32 @@ namespace igl
...
   //
-  template <typename MatV, typename MatF, typename Scalar>
+  template <typename MatV, typename MatF, typename MatK>
   IGL_INLINE void arap_linear_block(
     const MatV & V,
     const MatF & F,
     const int d,
     const igl::ARAPEnergyType energy,
-    Eigen::SparseMatrix<Scalar> & Kd);
+    MatK & Kd);
```
Comment (fwilliams) [inline include/igl/arap_linear_block.h]: Can these be either `SparseMatrixBase` or `SparseCompressedBase`?

Comment (fwilliams) [reply include/igl/arap_linear_block.h]: Note: I think we'll need to bump the version of Eigen for this.

File: include/igl/arap_rhs.cpp [hunk 4]
```diff
@@ -12,12 +12,13 @@
...
-  const Eigen::MatrixXd & V, 
-  const Eigen::MatrixXi & F,
-  const int dim,
-  const igl::ARAPEnergyType energy,
-  Eigen::SparseMatrix<double>& K)
+    const Eigen::MatrixBase<DerivedV> & V, 
+    const Eigen::MatrixBase<DerivedF> & F,
+    const int dim,
+    const igl::ARAPEnergyType energy,
+    Eigen::SparseMatrixBase<DerivedK>& K)
```
Comment (fwilliams) [inline include/igl/arap_rhs.cpp]: When do we use `SparseMatrixBase` and when do we use `SparseCompressedBase`?

File: include/igl/arap_rhs.h [hunk 5]
```diff
@@ -29,12 +29,13 @@ namespace igl
...
+  template<typename DerivedV, typename DerivedF, typename DerivedK>
   IGL_INLINE void arap_rhs(
-    const Eigen::MatrixXd & V, 
-    const Eigen::MatrixXi & F,
+    const Eigen::MatrixBase<DerivedV> & V, 
+    const Eigen::MatrixBase<DerivedF> & F,
     const int dim,
     const igl::ARAPEnergyType energy,
-    Eigen::SparseMatrix<double>& K);
+    Eigen::SparseMatrixBase<DerivedK>& K);
```
Comment (fwilliams) [inline include/igl/arap_rhs.h]: When do we use `SparseMatrixBase` and when do we use `SparseCompressedBase`?

File: include/igl/colon.cpp [hunk 6]
```diff
@@ -17,7 +17,7 @@ IGL_INLINE void igl::colon(
   const H hi,
```
Comment (fwilliams) [inline include/igl/colon.cpp]: We don't have to do this, but these should all be `Eigen::Index`

File: include/igl/vertex_components.cpp [hunk 7]
```diff
@@ -10,9 +10,9 @@
 #include <queue>
 #include <vector>
 
-template <typename AScalar, typename DerivedC, typename Derivedcounts>
+template <typename SparseT, typename DerivedC, typename Derivedcounts>
 IGL_INLINE void igl::vertex_components(
-  const Eigen::SparseMatrix<AScalar> & A,
+  const SparseT & A,
```
Comment (fwilliams) [inline include/igl/vertex_components.cpp]: `SparseCompressedBase`

File: include/igl/vertex_components.cpp [hunk 8]
```diff
@@ -67,9 +67,9 @@ IGL_INLINE void igl::vertex_components(
   }
 }
 
-template <typename AScalar, typename DerivedC>
+template <typename SparseT, typename DerivedC>
 IGL_INLINE void igl::vertex_components(
-  const Eigen::SparseMatrix<AScalar> & A,
+  const Eigen::SparseCompressedBase<SparseT> & A,
```
Comment (fwilliams) [inline include/igl/vertex_components.cpp]: :+1:

File: include/igl/arap_linear_block.h (hunk 3 above)
Comment (teseoch) [reply include/igl/arap_linear_block.h]: doesnt work. `SparseCompressedBase` does not support resize reserve etc

File: .travis.yml [hunk 9]
```diff
@@ -67,8 +57,5 @@ script:
 - make -j 2
 - ctest --verbose
 - popd
-- pushd python/tutorial
-- ${PYTHON} 101_FileIO.py
```
Comment (jdumas) [inline .travis.yml]: Why are you not running the python tutorial anymore?

File: cmake/libigl.cmake [hunk 10]
```diff
@@ -197,11 +196,7 @@ if(LIBIGL_WITH_CGAL)
     if(EXISTS ${LIBIGL_EXTERNAL}/boost)
       set(BOOST_ROOT "${LIBIGL_EXTERNAL}/boost")
     endif()
-    if(LIBIGL_WITH_PYTHON)
-      option(CGAL_Boost_USE_STATIC_LIBS "Use static Boost libs with CGAL" OFF)
-    else()
-      option(CGAL_Boost_USE_STATIC_LIBS "Use static Boost libs with CGAL" ON)
```
Comment (jdumas) [inline cmake/libigl.cmake]: Why remove this? You should leave one default option at least.

**Comment by fwilliams** [comment]
@jdumas yes. Right now we're working on the new python bindings in a separate repo which has libigl as a cmake external project. We're making changes to libigl and sending PRs (like this one) as we go along. 

@teseoch ran into some tricky compilation issues with the old bindings. We realized it would be much less effort if we killed the old bindings in this repository while we make changes necessary for the new bindings (as opposed to fixing the old bindings to work with the new changes). 

If others are using the python bindings in the dev branch, then we could move the old python bindings to a seperate repo with libigl as a CMake external project and point the external project version to the commit before this one.

**Comment by jdumas** [comment]
People are still using the `dev` branch with the old python bindings. The old python bindings are currently broken in `master`, and I think we should maintain a version of libigl with the old bindings until the new ones are ready for general use.

Why do the new python bindings use `libigl` as an external project? Wouldn't it be better to have them in the libigl main repo (we had this discussion before), as it would prevent them from using a deprecated version of libigl and make sure new PRs are not breaking existing python bindings.

**Comment by jdumas** [comment]
If anything, I would be more comfortable pushing the current `dev` branch into master before we merge this commit, and maybe create a release branch if we need to backport fixes until the new bindings are ready.

**Comment by fwilliams** [comment]
We decided with @danielepanozzo and @teseoch to have the python bindings in a separate repository while we develop them. This let's us "move fast and break things" in the binding code as well as have a small group of people have direct push access for bindings. Necessary changes to libigl are sent as pull requests as we go along and if we merge things fast enough, then we can keep everything up to date. 

In contrast if we were to directly develop against dev, every small change to the bindings will require a PR and a lot more overhead. 

We can decide later if we want to move the bindings into the main repo or not (there are pros and cons both ways and I don't really have an opinion on this tbh), but at least for now the setup we have makes sense if we want to get things done.

Pushing the current dev into master sounds like a great idea if its ready. How quickly can we do this by? I would really like to land this pull request so we can send more of them as we build out the bindings.

**Comment by jdumas** [comment]
Well, this depends on @alecjacobson and @danielepanozzo more than me. But before merging `dev` into master, we should take a look at other existing -- smaller -- pull-requests, and see if they can be merged. Then we need to update the changelog on the website with the list of new features, as we merge into master and tag the new release.

I thought we had created the `dev-python` exactly for the purpose of developing the new python bindings into libigl, since it is not protected, and can be kept in sync with master/dev as needed. It would be weird to turn `dev` into an unstable branch were python bindings are nuked until new ones are created. At least I'm worried it might give the wrong impression if this drags on.

File: cmake/libigl.cmake (hunk 10 above)
Comment (teseoch) [reply cmake/libigl.cmake]: i was too quick, fixed

File: .travis.yml (hunk 9 above)
Comment (teseoch) [reply .travis.yml]: python is gone

**Comment by teseoch** [comment]
something is wierd with appveyor, any idea?

**Comment by jdumas** [comment]
It probably means you need to resolve the conflicts against `dev` before AppVeyor can build again.

**Pull request closed without merging**